doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,898 @@
+    * The user will powerup/boot the DPUs as needed.
+
+    #### SmartSwitch Expected PowerDown sequence:
+    * When the smartswitch device is completely shutdown, it will try to gracefully shutdown all the DPUs first. In the event of failure to graceful shutdown will force shutdown.
+    * Then the switch will power down itself gracefully (NPU/Chassis)
+
+* The DPU must provide additional information such as reboot cause, timestamp, etc as explained in the scheme once it boots its OS to DPU_STATE table.
+* When the DPU reboots itself, should log the reboot cause and update the previous-reboot-cause field in the ChassisStateDB when it boots up again
+* The reboot-cause history should provide a holistic view of the reboot cause of the SmartSwitch host CPU, the reboot-cause of all the DPUs
+* The DPUs should be uniquely identified and the DPU upon boot may get this ID from the host and identify itself.
+* Implement the required API enhancements and new APIs for DPU management (see details in design section)
+* SmartSwitch should use the existing SONiC midplane-interface model in modular chassis design for communication between the DPU and the NPU
+* SmartSwitch should extend the SONiC modular chassis design and treat the dpu-cards just like line-cards in existing design
+* Reboot
+    * Only cold reboot of DPUs is required, warm boot support is not required.
+### 2.2. Monitoring and Thermal Management
+* Dpu State
+    * The DPUs should provide their state to the host as the boot progression happens by updating the dpu state data in the DPU_STATE table in the host ChassisStateDB (explained in DB schema)
+    * DPUs should be able to store the data using a redis call
+    * The DPU must provide the state information once it boots its OS to DPU_STATE table.
+    * The SmartSwitch host PMON should be able to monitor the liveliness of the DPUs and when they go down should be able to take appropriate actions  and should try to gracefully recover the DPU when requested by the PMON
+
+* Thermal management
+    * Besides additional DPU specific sensors, cooling device changes the logic remains the same.
+    * Sensor values, fan speeds and fan status should be read periodically and stored in SmartSwitch StateDB
+    * Platform modules should use the thermal sensor values against the thresholds in the thermal policy and adjust fan speeds depending on the temperature
+    * Trigger thermal shut down on critical policy violation
+
+* Show CLIs
+    * Extend existing CLIs such as 'show platform fan/temperature' to support the new HW
+    * Extend the modular chassis CLI 'show chassis modules status" to display the DPU state and health. (See CLIs section)
+
+### 2.3. Detect and Debug
+* Health
+    * SmartSwitch DPUs should store their health data in their local StateDB 
+    * DPUs should support a CLI to display the health data “show chassis health-events”
```

Comment:
@rameshraghupathy can we reuse existing `show system-health summary`

Reply:
God find.  Forgot to clean it up.  It is cleaned up now.